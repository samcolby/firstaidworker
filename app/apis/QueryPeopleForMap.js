import React from "react";
import PropTypes from "prop-types";

import { Query } from "react-apollo";

// Generate the GQL queries used in this component
import { QUERY_PEOPLE_FOR_MAP } from "../GraphQLQueries";

/**
 * Apollo Query to return a list of people to display on a map
 * @param {Object} props - Standard react prop
 * @param {function} children - Standard react prop, doesn't need to be included
 */
function QueryPeopleForMap({ children }) {
  return (
    <Query query={QUERY_PEOPLE_FOR_MAP} variables={{ limit: 50 }}>
      {props => {
        return children(props);
      }}
    </Query>
  );
}

QueryPeopleForMap.propTypes = {
  children: PropTypes.any
};

export default QueryPeopleForMap;
