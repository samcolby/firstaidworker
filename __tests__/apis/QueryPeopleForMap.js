import React from "react";
import { Text } from "react-native";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import { MockedProvider } from "react-apollo/test-utils";

import wait from "waait";

import QueryPeopleForMap from "../../app/apis/QueryPeopleForMap";

import {
  mocksForQueryPeopleForMap,
  mocksForQueryPeopleForMapError
} from "../mocksdata/MocksQueryPeopleForMap";

describe("<QueryPeopleForMap /> component", () => {
  it("Renders correctly", () => {
    const component = shallow(<QueryPeopleForMap />);
    expect(component.length).toBe(1);
  });

  it("Renders loading correctly", () => {
    const component = renderer.create(
      <MockedProvider mocks={mocksForQueryPeopleForMap} addTypename={false}>
        <QueryPeopleForMap>
          {({ data, error, fetchMore, loading, networkStatus, refetch }) => {
            if (error) {
              return <Text>error</Text>;
            } else if (loading) {
              return <Text>Loading</Text>;
            } else {
              return <Text>{JSON.stringify(data.workers)}</Text>;
            }
          }}
        </QueryPeopleForMap>
      </MockedProvider>
    );

    const textComponent = component.root.findByType("Text");
    expect(textComponent.children[0]).toEqual("Loading");
    expect(component).toMatchSnapshot();
  });

  it("Renders data correctly", async () => {
    const component = renderer.create(
      <MockedProvider mocks={mocksForQueryPeopleForMap} addTypename={false}>
        <QueryPeopleForMap>
          {({ data, error, fetchMore, loading, networkStatus, refetch }) => {
            if (error) {
              return <Text>Error: {JSON.stringify(error)}</Text>;
            } else if (loading) {
              return <Text>loading</Text>;
            } else {
              return <Text>{JSON.stringify(data.workers)}</Text>;
            }
          }}
        </QueryPeopleForMap>
      </MockedProvider>
    );

    await wait(0);

    const textComponent = component.root.findByType("Text");
    expect(JSON.parse(textComponent.children)[0].name).toEqual(
      "Kaitlin Burnett"
    );

    expect(component).toMatchSnapshot();
  });

  it("Throws errors", async () => {
    const component = renderer.create(
      <MockedProvider
        mocks={mocksForQueryPeopleForMapError}
        addTypename={false}
      >
        <QueryPeopleForMap>
          {({ data, error, fetchMore, loading, networkStatus, refetch }) => {
            if (error) {
              return <Text>ERROR!</Text>;
            } else if (loading) {
              return <Text>loading</Text>;
            } else {
              return <Text>{JSON.stringify(data.workers)}</Text>;
            }
          }}
        </QueryPeopleForMap>
      </MockedProvider>
    );

    await wait(0); // wait for response

    const textComponent = component.root.findByType("Text");
    expect(textComponent.children[0]).toContain("ERROR!");
  });
});
