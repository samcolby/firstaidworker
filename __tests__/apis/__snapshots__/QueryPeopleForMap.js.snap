// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QueryPeopleForMap /> component Renders data correctly 1`] = `
<Text>
  [{"id":"9f47bfa4-892e-4abd-8a18-0576f8759051","name":"Kaitlin Burnett","job_title":"Marketing","avatar_uri":"https://randomuser.me/api/portraits/women/17.jpg","location":{"coordinates":[51.1910548,-4.1274755]}},{"id":"0b8a6373-4530-41d7-87a9-435552dfa334","name":"Sargent Bush","job_title":"Sales","avatar_uri":"https://randomuser.me/api/portraits/men/32.jpg","location":{"coordinates":[51.2096548,-4.1144755]}},{"id":"54f58dfd-3f17-427c-b1d1-74c26a77f700","name":"Key Whitley","job_title":"Michael Jackson Impersonator","avatar_uri":"https://randomuser.me/api/portraits/women/45.jpg","location":{"coordinates":[51.1826548,-4.0674755]}}]
</Text>
`;

exports[`<QueryPeopleForMap /> component Renders loading correctly 1`] = `
<Text>
  Loading
</Text>
`;
